### PR TITLE
Fixed the compiler.jl and inlining algorithm demos in documentation

### PR DESCRIPTION
**Pull request by vjhaanpaa**
Issue #34682 noted that the below demo from [Inference documentation](https://docs.julialang.org/en/v1/devdocs/inference/) gives a `MethodError`:

```julia
# Get the method
atypes = Tuple{Type{Int}, UInt}  # argument types
mths = methods(convert, atypes)  # worth checking that there is only one
m = first(mths)

# Create variables needed to call `typeinf_code`
params = Core.Compiler.Params(typemax(UInt))  # parameter is the world age,
                                                        #   typemax(UInt) -> most recent
sparams = Core.svec()      # this particular method doesn't have type-parameters
optimize = true            # run all inference optimizations
cached = false             # force inference to happen (do not use cached results)
Core.Compiler.typeinf_code(m, atypes, sparams, optimize, cached, params)
```

The specific error message is what is shown in the `REPL` when the last line is run, which complains about an additional `::Bool` in `Core.Compiler.typeinf_code()`: 

```julia
julia> Core.Compiler.typeinf_code(m, atypes, sparams, optimize, cached, params)
ERROR: MethodError: no method matching typeinf_code(::Method, ::Type{Tuple{Type{Int64},UInt64}}, ::Core.SimpleVector, ::Bool, ::Bool, ::Core.Compiler.Params)
Closest candidates are:
  typeinf_code(::Method, ::Any, ::Core.SimpleVector, ::Bool, ::Core.Compiler.Params) at compiler/typeinfer.jl:510
Stacktrace:
 [1] top-level scope at REPL[8]:1
```

In the issue, @GiggleLiu guesses that the API has been changed, and gives the below example for the latest API, in which the keyword `cached` is dropped from `Core.Compiler.typeinf_code()`.

```julia
julia> Core.Compiler.typeinf_code(m, atypes, sparams, optimize, params)
(CodeInfo(
    @ number.jl:7 within `convert'
1 ─     ($(Expr(:static_parameter, 1)))(x)::Union{}
└──     $(Expr(:unreachable))::Union{}
), Union{})
```

Dropping `cached` fixes the demo for `compiler.jl` in the documentation, and this pull request makes that change to `inference.md`. 

Additionally, the presented output of the inlining algorithm demo in the same file was corrected.

### CODE REVIEW

File: doc/src/devdocs/inference.md [hunk 1]
```diff
@@ -109,26 +108,11 @@ cst = map(cost, ci.code)
 
 # output
 
-31-element Array{Int64,1}:
```
Comment (KristofferC) [inline doc/src/devdocs/inference.md]: Why change this?

Comment (vjhaanpaa) [reply doc/src/devdocs/inference.md]: If you run the demo, you get only a 5-element array. Try running this in `REPL`: 

```julia
# A demo on `fill(3.5, (2, 3))
f = fill
tt = Tuple{Float64, Tuple{Int,Int}}
# Create the objects we need to interact with the compiler
params = Core.Compiler.Params(typemax(UInt))
mi = Base.method_instances(f, tt)[1]
ci = code_typed(f, tt)[1][1]
opt = Core.Compiler.OptimizationState(mi, params)
# Calculate cost of each statement
cost(stmt::Expr) = Core.Compiler.statement_cost(stmt, -1, ci, opt.sptypes, opt.slottypes, opt.params)
cost(stmt) = 0
cst = map(cost, ci.code)
```

The current documentation lists a five-element array, and I am not entirely sure why it has been changed to a 31-element array. If I am wrong, the second demo can be just ignored, but the `doctest` failed there as well due to a reason that I don't 100% understand.

Comment (KristofferC) [reply doc/src/devdocs/inference.md]: You might (likely) run it on a version of Julia that is different from the one where the doctests are run on (master).

File: doc/src/devdocs/inference.md [hunk 2]
```diff
@@ -31,8 +31,7 @@ params = Core.Compiler.Params(typemax(UInt))  # parameter is the world age,
                                                         #   typemax(UInt) -> most recent
 sparams = Core.svec()      # this particular method doesn't have type-parameters
```
Comment (vtjnash) [inline doc/src/devdocs/inference.md]:
```suggestion
sparams = Core.svec(Int64)      # this particular method has a type-parameter
```

File: doc/src/devdocs/inference.md [hunk 3]
```diff
@@ -31,8 +31,7 @@ params = Core.Compiler.Params(typemax(UInt))  # parameter is the world age,
                                                         #   typemax(UInt) -> most recent
```
Comment (vtjnash) [inline doc/src/devdocs/inference.md]: This is wrong now—the `typemax(UInt)` world doesn't exist.
```suggestion
params = Core.Compiler.Params(Base.get_world_counter())
```

File: doc/src/devdocs/inference.md (hunk 2 above)
Comment (vtjnash) [reply doc/src/devdocs/inference.md]: Running this code sample would previously have permanently corrupted the internal state of the system due to this

**Comment by vtjnash** [comment]
Thanks for looking into this! There was even more problems as well, which I've noted the corrections for.

**Comment by vjhaanpaa** [comment]
> Thanks for looking into this! There was even more problems as well, which I've noted the corrections for.

Thanks for your feedback. I looked into the suggestion by @KristofferC about running the second demo on an out-of-date version, and I had indeed mixed my terminal windows and `julia` instances. I'll pull together the changes and make a new commit.

**Comment by vjhaanpaa** [comment]
Okay, I'm now certainly running `Version 1.5.0-DEV.339 (2020-02-25)` and there are still discrepancies with the second demo. Any suggestions?

**Comment by ViralBShah** [comment]
@KristofferC Is this ok to merge?

**Comment by ViralBShah** [comment]
Bump @KristofferC or should it be someone else?

**Pull request merged**